src/order/boolean_algebra.lean

```diff
@@ -840,6 +840,9 @@ instance : boolean_algebra bool := boolean_algebra.of_core
   top_le_sup_compl := λ a, a.bor_bnot_self.ge,
   ..bool.linear_order, ..bool.bounded_order }
 
+instance is_irrefl.compl (r) [is_irrefl α r] : is_refl α rᶜ := ⟨@irrefl α r _⟩
+instance is_refl.compl (r) [is_refl α r] : is_irrefl α rᶜ := ⟨λ a, not_not_intro (refl a)⟩
```

Comment:
These seem to make sense as instances.

It might be organizationally better if these come right after the `pi.has_compl` instance, rather than right after "Yael's theorem" (the resolution to the long-standing conjecture of whether `bool` was a boolean algebra).